### PR TITLE
Rails 4 upgrade Fixes #57 

### LINKED ISSUE #57: Upgrade to Rails 4.x
We can't upgrade to rails 4 until we do other infrastructure upgrades first (see: #40 and #39).

### PR DESCRIPTION
**Pull request by tlynam**
Upgraded gems, scopes, configs, params, queries, suggestion test, match routes

Followed
http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-3-2-to-rails-4-0
http://railscasts.com/episodes/415-upgrading-to-rails-4

### CODE REVIEW

File: app/models/member.rb [hunk 1]
```diff
@@ -15,9 +15,9 @@ class Member < ActiveRecord::Base
   validates :ruby_gems_id, :rubygems => true, :allow_blank => true, uniqueness: true
   validates :website,      :url => true,      :allow_blank => true, uniqueness: true
 
-  scope :featured, where(featured: true)
-  scope :regular, where(featured: false)
-  scope :verified, where(verified: true)
+  scope :featured, -> { where(featured: true) }
+  scope :regular, -> { where(featured: false) }
+  scope :verified, -> { where(verified: true) }
```
Comment (zenspider) [inline app/models/member.rb]: call me old fashioned but... I _hate_ stabby proc.

File: config/initializers/secret_token.rb [hunk 2]
```diff
@@ -5,3 +5,4 @@
 # Make sure the secret is at least 30 characters and all random,
 # no regular words or you'll be exposed to dictionary attacks.
 SeattlerbOrg::Application.config.secret_token = 'eff82d437b2886034bb8da23aa99d75499baeed3f0ab1864774fdfaf66b98a935a1bb54377ae05c927af8eecba325e9cba80aec3cb5818c95a8da71539184186'
+SeattlerbOrg::Application.config.secret_key_base = '55a89dc06da0cc081f107e6702d58ea8e0c94b78903af52b5a791dad348619b740a0ceb36d575f5dafca6b6a9ae79ee37c988e11b2d707d657550b70a9d1f3aa'
```
Comment (zenspider) [inline config/initializers/secret_token.rb]: guessing these shouldn't be checked in?

File: config/initializers/devise.rb [hunk 3]
```diff
@@ -13,4 +13,6 @@
 
   # rememberable
   config.remember_for = 12.weeks
+
+  config.secret_key = '966d17ce12454589478a70cffa56b8407e59acae8cc8aed66e1bb05b96527c38ccf2fd87e8337ad7430d9bd5fcbcf618f4cefb26b56f0d142398855084323483'
```
Comment (zenspider) [inline config/initializers/devise.rb]: not so secret key?

File: test/models/suggestion_test.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 class SuggestionTest < ActiveSupport::TestCase
   def test_valid
-    suggestion = Suggestion.new(:title => "WOOT", :suggester => "Fred")
+    suggestion = Suggestion.new(title: "WOOT", suggester: "Fred", comment: "")
```
Comment (zenspider) [inline test/models/suggestion_test.rb]: style changes should be in a separate PR

File: test/controllers/suggestions_controller_test.rb [hunk 5]
```diff
@@ -29,7 +29,7 @@ def test_new
 
   def test_create
     assert_difference('Suggestion.count', 1) do
-      post :create, suggestion: {:title => "My title", :suggester => "Barney Rubble" }
+      post :create, suggestion: {title: "My title", suggester: "Barney Rubble", comment: "" }
```
Comment (zenspider) [inline test/controllers/suggestions_controller_test.rb]: style changes should be in a separate PR

**Comment by zenspider** [comment]
The rest of this looks fine to me, but I'd prefer a confirmation from both @tenderlove and either @phiggins or @thagomizer (not sure she currently CAN sign off on things, legally--that might be considered IP :P).

**Comment by zenspider** [comment]
The tests fail on the CI... which is odd because you can see it failing for a column when the migration clearly ran.

File: config/initializers/secret_token.rb (hunk 2 above)
Comment (tenderlove) [inline config/initializers/secret_token.rb]: Ya, that's right.  We shouldn't be checking these in.  It's OK for dev and test envs, but for production, we should be reading these from an environment variable on Heroku.

**Comment by tenderlove** [comment]
It looks good except for the secret token part.  We [should be using an environment variable on heroku](http://daniel.fone.net.nz/blog/2013/05/20/a-better-way-to-manage-the-rails-secret-token/) to manage that.

**Comment by TLynam** [comment]
Thanks a lot!  We'll need to add these secret environment variables to Heroku and looks like Travis CI
ENV["SECRET_KEY"]
ENV["SECRET_TOKEN"]
ENV["SECRET_KEY_BASE"]

**Comment by phiggins** [comment]
Except for the secret keys, this looks good to me. We should probably add parameter sanitization around everything we accept as user input (talks, users, ???), but that could be another PR.

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -13,4 +13,6 @@
 
   # rememberable
   config.remember_for = 12.weeks
+
+  config.secret_key = ENV["SECRET_KEY"]
```
Comment (phiggins) [inline config/initializers/devise.rb]: To make this easier to use in dev, it might make sense to do something like:

```
config.secret_key = if Rails.env.production?
  ENV.fetch("SECRET_KEY")
else
  ENV.fetch("SECRET_KEY", "secretkey")
end
```

I'm not sure if this is a terrible idea or not.

**Comment by tlynam** [comment]
Hi @zenspider and @phiggins, is it alright if we remove this update all code in the completed talks migration so tests pass?  Is it needed if the field has a default value?

**Comment by phiggins** [comment]
> Hi @zenspider and @phiggins, is it alright if we remove this update all code in the completed talks migration so tests pass? Is it needed if the field has a default value?

I think this is fine. Anyone with an existing database might get into a weird state, but it should be an easy fix.
